src/Symfony/Component/HttpKernel/Controller/ControllerResolverInterface.php

```diff
@@ -42,16 +42,4 @@
      * @throws \LogicException If the controller can't be found
      */
     public function getController(Request $request);
-
-    /**
-     * Returns the arguments to pass to the controller.
-     *
-     * @param Request  $request    A Request instance
-     * @param callable $controller A PHP callable
-     *
-     * @return array An array of arguments to pass to the controller
-     *
-     * @throws \RuntimeException When value for argument given is not provided
-     */
-    public function getArguments(Request $request, $controller);
```

Comment:
@fabpot would you want the interfaces extending each other? 
 - This would cause a very annoying BC break in 4.0 
 - and having users implement a method that shouldn't ever be called. 

Or do you want to have:
 - duck-typing in the edge cases where people wrote a whole new `ControllerResolver` 
 - _and_ manually called this (highly unlikely)
 -  and a clearly separated interface which causes no migration issues in 4.0

I prefer to leave it as is and take the "break" for granted, if you can really call it a break.